### PR TITLE
add `paddle.nn.loss.CrossEntropyLoss` op

### PR DESCRIPTION
**Pull request by yghstill**
test=develop

Add a new public nn API，`paddle.nn.loss.CrossEntropyLoss(reduction='sum'|'mean'|'none')`.

Usage example:

```python
# declarative mode
import paddle
import paddle.fluid as fluid
import numpy as np

input = fluid.layers.data(name='input', shape=[3,5], dtype='float32')
label = fluid.layers.data(name='label', shape=[3,1], dtype='int64')
weight = fluid.layers.data(name='weight', shape=[5], dtype='float32')
ce_loss = paddle.nn.loss.CrossEntropyLoss(weight=weight, reduction='mean')
output = ce_loss(input,label)
place = fluid.CPUPlace()
exe = fluid.Executor(place)
exe.run(fluid.default_startup_program())
input_data = np.random.random([3,5]).astype("float32")
label_data = np.array([[1], [3], [4]]).astype("int64")
weight_data = np.random.random([5]).astype("float32")
output = exe.run(fluid.default_main_program(),
              feed={"input":input_data, "label":label_data,"weight":weight_data},
              fetch_list=[output],
              return_numpy=True)
 print(output)

# imperative mode
import paddle.fluid.dygraph as dg
with dg.guard(place) as g:
    input = dg.to_variable(input_data)
    label = dg.to_variable(label_data)
    weight = dg.to_variable(weight_data)
    ce_loss = paddle.nn.loss.CrossEntropyLoss(weight=weight, reduction='mean')
    output = ce_loss(input,label)
    print(output.numpy())

### CODE REVIEW

**Review by heavengate — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+import paddle
+import paddle.fluid as fluid
+import numpy as np
+import unittest
+
+
+class CrossEntropyLoss(unittest.TestCase):
+    def test_cross_entropy_loss_mean(self):
+        input_np = np.random.random([3, 5]).astype(np.float32)
```
Comment (qingqing01) [inline python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py]: See:

https://github.com/PaddlePaddle/Paddle/wiki/OP-test-input-shape-requirements

the shape in unit testing is too small

File: python/paddle/nn/layer/loss.py [hunk 2]
```diff
@@ -16,14 +16,124 @@
...
     #    'MSELoss',
     'L1Loss',
     #    'NLLLoss',
     #   'BCELoss'
 ]
 
 
+class CrossEntropyLoss(fluid.dygraph.Layer):
+    """
+    This operator implements the cross entropy loss function.
```
Comment (qingqing01) [inline python/paddle/nn/layer/loss.py]: 1. This `CrossEntropyLoss`, combile softmax, cross_entropy, and reduce_sum/mean together, need to describe more clearly.
2. Wheteher this class is the same with PyTorch?   Pytorch combines `log_softmax` and `nll_loss`.

File: python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                weight=fluid.dygraph.to_variable(weight_np))
+            dy_ret = cross_entropy_loss(
+                fluid.dygraph.to_variable(input_np),
+                fluid.dygraph.to_variable(label_np))
+            dy_ret_value = dy_ret.numpy()
+            self.assertIsNotNone(dy_ret_value)
+        self.assertTrue(np.allclose(static_ret, dy_ret_value))
+
+    def test_cross_entropy_loss_sum(self):
+        input_np = np.random.random([3, 5]).astype(np.float32)
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py]: @zhupengyang 能否看下这个shape不符合大于100的要求，但CI没有拦截的问题呢？

Comment (zhupengyang) [reply python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py]: 用unittest.TestCase直接写的都是检测不到的，只能检测继承自OpTest的单测

File: python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py (hunk 1 above)
Comment (yghstill) [reply python/paddle/fluid/tests/unittests/test_cross_entropy_loss.py]: DONE

File: python/paddle/nn/layer/loss.py (hunk 2 above)
Comment (yghstill) [reply python/paddle/nn/layer/loss.py]: DONE

**Review by qingqing01 — APPROVED** [review]

**Review by heavengate — APPROVED** [review]
LGTM
